models/OrderPickupEventModel.ts

```diff
@@ -23,13 +23,15 @@ export class OrderPickupEventModel extends BaseEntity {
   @JoinColumn({ name: 'order' })
   orders: OrderModel[];
 
-  public getPublicOrderPickupEvent(): PublicOrderPickupEvent {
-    return {
+  public getPublicOrderPickupEvent(canSeeOrders = false): PublicOrderPickupEvent {
+    const pickupEvent: PublicOrderPickupEvent = {
       uuid: this.uuid,
       title: this.title,
       start: this.start,
       end: this.end,
       description: this.description,
     };
+    if (canSeeOrders) pickupEvent.orders = this.orders.map((order) => order.getPublicOrder());
```

Comment:
Sorry, I'm not quite catching what you're saying here. Can you explain this again?

Reply:
1. If you call the `GET /merch/order/:uuid` endpoint, then you find an `OrderModel`.
2. The `OrderModel` has `eager: true` set for the pickup event so it includes `pickupEvent: OrderPickupEventModel`, but the `OrderPickupEventModel` that's returned as part of the `OrderModel` doesn't have `orders: OrderModel[]`, right? The `OrderPickupEventModel` wouldn't have an `orders` field at all since there's no join on that. So `orders` might be null or undefined.
3. When `OrderModel.getPublicOrder()`'s called, `OrderPickupEventModel.getPublicPickupEvent()`'s also called, right?
3. So, if `orders` is null or undefined, `this.orders.map(...)` throws some sort of exception, wouldn't it?

That's the issue I ran into a while back, where some `getPublic` call to one model triggered a `getPublic` call to another model that didn't have some other `Model[]`.

Side note: should `GET /order` be `GET /orders` instead? I think so.

Reply:
Hope that's clearer, it's kind of a weird case. If it still doesn't make sense, I can pull your branch and test it myself.

Reply:
Ah okay got it, thanks for the detail. I think this isn't currently an issue though, because `OrderModel.getPublicOrder()` calls `OrderPickupEventModel.getPublicPickupEvent()` with `canSeeOrders = false` (the default boolean value when nothing is passed), so the `this.orders.map()` check wouldn't be ran since it's false.

The only case I can think of where this line would throw potentially an NPE is when `getPublicPickupEvent` is called directly on the pickup event that hasn't been queried from the database (i.e., it's been created from the merch factory, because otherwise the `orders` join happens), but even then it wouldn't because the merch factory sets `orders: []` in the pickup event creation.